.github/workflows/run-tests.yml

```diff
@@ -85,16 +85,55 @@ jobs:
         if: ${{ always() }}
         id: cli-messaging-tests
         run: yarn test-cli-messaging
+      - if: ${{ steps.cli-messaging-tests.outcome != 'success' }}
+        uses: ./github-actions/summarize-errors
+        with:
+          location: "${{github.workspace}}/cli-messaging"
+          project-name: "cli-messaging"
+          artifact-name: cli-messaging-tests-${{ runner.os }}-${{ matrix.java }}-${{ matrix.node.artifact }}
+      - if: ${{ steps.cli-messaging-tests.outcome != 'success' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: cli-messaging-tests-${{ runner.os }}-${{ matrix.java }}-${{ matrix.node.artifact }}
```

Comment:
Possibly controversial change: Only tests that actually fail will produce artifacts, and those artifacts should have more specific names.

Reply:
Thanks for calling this out. Is there a specific reason behind not uploading artifacts for successful runs?

Reply:
Sorry, just now seeing this.
If we want, we could have the artifacts be uploaded for all tests. There's nothing stopping us from doing that. In my opinion, it is still cleaner for each test to have its own artifact instead of all results for a given run being bundled into one artifact. Thoughts?